src/main/java/ch/uzh/ifi/hase/soprafs21/controller/PathsApiController.java

```diff
@@ -11,14 +23,38 @@
 
     private final NativeWebRequest request;
 
+    private final UserService userService;
+    private final PathService pathService;
+    private GeometryFactory geometryFactory;
+
+
     @org.springframework.beans.factory.annotation.Autowired
-    public PathsApiController(NativeWebRequest request) {
+    public PathsApiController(NativeWebRequest request, UserService userService, PathService pathService, GeometryFactory geometryFactory) {
         this.request = request;
+        this.userService = userService;
+        this.pathService = pathService;
+        this.geometryFactory =geometryFactory;
     }
 
     @Override
     public Optional<NativeWebRequest> getRequest() {
         return Optional.ofNullable(request);
     }
 
+    @Override
+    public ResponseEntity<Void> addPath(@Valid WalkingRouteDto walkingRouteDto) throws Exception {
+        if (walkingRouteDto.getId() != null) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Id is not allowed in POST");
+        }
+        if (!userService.isRequesterAndAuthenticatedUserTheSame(walkingRouteDto.getCreator().getId())){
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN, "User is not permitted to add this Route");
+        }
+
+        Path path = PathDTOMapper.INSTANCE.toPathEntity(walkingRouteDto);
+        path.setCreator(userService.getUserById(walkingRouteDto.getCreator().getId()));
+        path.setRoute(geometryFactory.createLineString(SpatialDTOMapper.INSTANCE.getCoordinates(walkingRouteDto.getListOfCoordinates())));
```

Comment:
this is a part of Mapper logic. Mapping the route coordinate should be inside PathsDTOMapper.

Reply:
updated